data_structures/avltree_map.cpp

```diff
@@ -0,0 +1,226 @@
+/**
+ * @file avltree_map.cpp
+ * @brief Unit testings for avltree_map.hpp
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible) and a **detailed description**.
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂